aptos-move/block-executor/src/view.rs

```diff
@@ -221,76 +224,113 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> LatestView<
     }
 }
 
-impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> TStateView
+impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> TResourceView
     for LatestView<'a, T, S, X>
 {
     type Key = T::Key;
+    type Layout = MoveTypeLayout;
 
-    fn get_state_value(&self, state_key: &T::Key) -> anyhow::Result<Option<StateValue>> {
+    fn get_resource_state_value(
+        &self,
+        state_key: &Self::Key,
+        _maybe_layout: Option<&Self::Layout>,
+    ) -> anyhow::Result<Option<StateValue>> {
         match &self.latest_view {
-            ViewState::Sync(state) => match state_key.module_path() {
-                Some(_) => {
-                    use MVModulesError::*;
-                    use MVModulesOutput::*;
+            ViewState::Sync(state) => {
+                let mut mv_value = state.fetch_data(state_key, self.txn_idx);
 
-                    match state.fetch_module(state_key, self.txn_idx) {
-                        Ok(Executable(_)) => unreachable!("Versioned executable not implemented"),
-                        Ok(Module((v, _))) => Ok(v.as_state_value()),
-                        Err(Dependency(_)) => {
-                            // Return anything (e.g. module does not exist) to avoid waiting,
-                            // because parallel execution will fall back to sequential anyway.
-                            Ok(None)
-                        },
-                        Err(NotFound) => self.base_view.get_state_value(state_key),
-                    }
-                },
-                None => {
-                    let mut mv_value = state.fetch_data(state_key, self.txn_idx);
+                if matches!(mv_value, ReadResult::Unresolved) {
```

Comment:
i thought aggregator resolve goes to aggregator view instead of resource view? is that planned for later?